enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1319 @@
+---
+title: set_eip_nlb_ingress
+authors:
+  - miheer
+reviewers:
+  - Miciah
+  - frobware
+  - gspence
+  - candace
+approvers:
+  - Miciah
+  - frobware
+  - gspence
+api-approvers:
+  - joel
+  - deads
+creation-date: 2024-03-06
+last-updated: 2024-03-22
+tracking-link:
+  - "https://issues.redhat.com/browse/NE-1274"
+see-also:
+  - ""
+replaces:
+  - ""
+superseded-by:
+  - ""
+---
+
+# Set AWS EIP For NLB Ingress Controller
+
+## Summary
+
+This enhancement allows user to set AWS EIP for the NLB default or custom ingress controller.
+This is a feature request to enhance the IngressController API to be able to support static IPs during
+- Install time
+- Custom NLB ingress controller creation 
+- Reconfiguration of the custom and default router to NLB with EIP allocation.
+
+## Motivation
+
+### User Stories
+
+- As a cluster administrator, I want to provision EIPs and use them with an NLB IngressController.
+- As a cluster administrator, I want to ensure EIPs are used with NLB on default router at install time.
+- As a cluster administrator, I want to reconfigure default router to use EIPs.
+- As a cluster administrator of OpenShift on AWS (ROSA), I want to use static IPs (and therefore AWS Elastic IPs) so that 
+  I can configure appropriate firewall rules.
+  I want the default AWS Load Balancer that they use (NLB) for their router to use these EIPs.
+- As a cluster administrator who has purchased a block of public IPv4 addresses, I want to use these IP addresses, so 
+  that I can avoid Amazon's new charge for public IP addresses.
```

Comment:
Does EIP means always use External (Internet-facing) LB ? I'm wondering if Internal LB support EIPs

Reply:
> always use External (Internet-facing) LB ?

Short answer yes - for this use case. From [docs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-eips-allocating) state:

> An Elastic IP address is a public IPv4 address, which is reachable from the internet.

Private Network Load Balancers can assign private IP, but not through EIP. The [ELB docs][docs-lb] states:

> For internal load balancers, you can specify one private IP address per subnet from the IPv4 range of the subnet.


FWIW additionally, there is another use case using Outpost - not applied to this EP as it is out of the scope to support Outpost/ALB:
EIPs can assign IPs from a Customer Owned IP (CoIP) Pool when using Outposts, which can be a private IP from a customer, not from AWS boarder group. From [Customer-owned IP addresses][docs-op] docs, the example "Example: Internet connectivity through the on-premises networ" shares the use case of using an EIP allocated from CoIP with private IP:

> - The local gateway uses BGP advertisement to advertise the customer-owned IP address pool to the on-premises network.
 > - An Elastic IP address association that maps 10.0.3.112 to 10.1.0.2.

[docs-lb]: https://docs.aws.amazon.com/elasticloadbalancing/latest/APIReference/API_CreateLoadBalancer.html
[docs-eip]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-eips-allocating
[docs-op]: https://docs.aws.amazon.com/outposts/latest/userguide/routing.html#ip-addressing